cpp/src/arrow/array/statistics.h

```diff
@@ -47,14 +47,14 @@ struct ARROW_EXPORT ArrayStatistics {
   /// \brief The minimum value, may not be set
   std::optional<ValueType> min = std::nullopt;
 
-  /// \brief Whether the minimum value is exact or not, may not be set
-  std::optional<bool> is_min_exact = std::nullopt;
+  /// \brief Whether the minimum value is exact or not
```

Comment:
This looks good to me, but IMO `is_min_exact = false` might still has exact statistics but the reader cannot gurantee that, since https://github.com/apache/parquet-format/pull/216 is new in 2.10 :-(

Reply:
Are the followings correct?

1. Parquet 2.9 or earlier data don't have `is_min_value_exact`/`is_max_value_exact`
2. Parquet 2.9 or earlier data use only exact min/max
3. Parquet 2.10 or later data use exact min/max or non-exact min/max
4. Parquet 2.10 or later data may use exact min/max without `is_min_value_exact`/`is_max_value_exact`

You're focusing the 2. case, right? Can our Parquet reader detect Parquet version? If so, can we always set `true` to `is_min_exact`/`is_max_exact` for Parquet 2.9 or earlier?

Reply:
Sorry for late replying

> Parquet 2.9 or earlier data don't have is_min_value_exact/is_max_value_exact

Yes

> Parquet 2.9 or earlier data use only exact min/max

I guess no, I'll send a mail to maillist to make it sure

> Parquet 2.10 or later data use exact min/max or non-exact min/max

Yes

> Parquet 2.10 or later data may use exact min/max without is_min_value_exact/is_max_value_exact

right

Reply:
>  If so, can we always set true to is_min_exact/is_max_exact for Parquet 2.9 or earlier?

Hmmm I'll try to make it clear

Reply:
Thanks!

Reply:
Anyway we can mention "exact=false" can also means is exact, lol

Reply:
OK. Let's use `exact=true` for Apache Parquet C++ output.